### PR TITLE
Verify that all device paths passed to trait methods are absolute

### PR DESCRIPTION
**Pull request by mulkieran**
Closes: https://github.com/stratis-storage/project/issues/229

### CODE REVIEW

**Comment by mulkieran** [comment]
I think that this will do. But I updated an old issue: https://github.com/stratis-storage/stratisd/issues/385. It would be better to rely on our source to enforce what we need, but the proposed change would be quite invasive, so best to defer it.

File: src/engine/shared.rs [hunk 1]
```diff
@@ -162,3 +162,116 @@ pub fn set_key_shared(key_fd: RawFd, interactive: Option<bool>) -> StratisResult
...
+
+/// Validate a str for use as a Pool or Filesystem name.
+pub fn validate_name(name: &str) -> StratisResult<()> {
+    let name_path = Path::new(name);
+    if name.contains('\u{0}') {
+        return Err(StratisError::Engine(
+            ErrorEnum::Invalid,
+            format!("Name contains NULL characters : {}", name),
+        ));
+    }
```
Comment (jbaublitz) [inline src/engine/shared.rs:174]: Is there a reason that this doesn't come before `let name_path ...`?

Comment (mulkieran) [reply src/engine/shared.rs:174]: I doubt it! I didn't look inside this method when I moved it. I'll swap the order.

Comment (jbaublitz) [reply src/engine/shared.rs:174]: Perfect, thanks!

**Review by jbaublitz — APPROVED** [review]

**Pull request merged**